Source/Core/DolphinQt2/GameList/GameFile.cpp

```diff
@@ -427,3 +387,92 @@ QString FormatSize(qint64 size)
   }
   return QStringLiteral("%1 %2").arg(QString::number(num, 'f', 1)).arg(unit);
 }
+
+template <typename T, typename U = std::enable_if_t<std::is_enum<T>::value>>
+QDataStream& operator<<(QDataStream& out, const T& enum_value)
+{
+  out << static_cast<std::underlying_type_t<T>>(enum_value);
+  return out;
+}
+
+template <typename T, typename U = std::enable_if_t<std::is_enum<T>::value>>
+QDataStream& operator>>(QDataStream& in, T& enum_value)
+{
+  std::underlying_type_t<T> tmp;
+  in >> tmp;
+  enum_value = static_cast<T>(tmp);
+  return in;
+}
+
+// Some C++ implementations define uint64_t as an 'unsigned long', but QDataStream only has built-in
+// overloads for quint64, which is an 'unsigned long long' on Unix
+QDataStream& operator<<(QDataStream& out, const unsigned long& integer)
+{
+  out << static_cast<quint64>(integer);
+  return out;
+}
+QDataStream& operator>>(QDataStream& in, unsigned long& integer)
+{
+  quint64 tmp;
+  in >> tmp;
+  tmp = integer;
+  return in;
+}
+
+QDataStream& operator<<(QDataStream& out, const GameFile& file)
+{
+  out << file.m_last_modified;
```

Comment:
These also need to store/load `m_size` and `m_issues`. (If you want to go above and beyond, it'd be nice to have the members listed in the same order as in the class definition, to make it easy to compare.)